# Q1：

# (a) Solution: One-Sample t-test

### State the Hypotheses
- Null hypothesis ($H_0$): The mean part length equals the standard value.
  $H_0: \mu = 10.0 \, \text{cm}$
- Alternative hypothesis ($H_1$): The mean part length is significantly different from the standard value.
  $H_1: \mu \neq 10.0 \, \text{cm}$ (Two-tailed test)

In [1]:
# Given data
sample_mean = 10.06  # Sample mean (cm)
pop_mean = 10.0      # Hypothesized population mean (cm)
sample_std = 0.19    # Sample standard deviation (cm)
n = 25               # Sample size
alpha = 0.05         # Significance level

# Calculate test statistic (t-value)
import numpy as np
t_stat = (sample_mean - pop_mean) / (sample_std / np.sqrt(n))

# Calculate critical t-value
from scipy import stats
df = n - 1  # Degrees of freedom
t_critical = stats.t.ppf(1 - alpha/2, df)  # Two-tailed critical value

print(f'Test statistic (t): {t_stat:.3f}')
print(f'Degrees of freedom: {df}')
print(f'Critical t-values: ±{t_critical:.3f}')

Test statistic (t): 1.579
Degrees of freedom: 24
Critical t-values: ±2.064


### Rejection Region
The rejection region for a two-tailed test with $\alpha = 0.05$ and $df = 24$ is:
$t < -2.064$ or $t > 2.064$

### Conclusion
The calculated test statistic ($t = 1.579$) falls within the non-rejection region ($-2.064 < 1.579 < 2.064$).

We fail to reject the null hypothesis $H_0$. At the $\alpha = 0.05$ significance level, there is insufficient evidence to conclude that the mean part length differs from the standard value of 10.0 cm.

## (b): p-value Calculation and Interpretation

In [3]:
# Use previously defined variables (t_stat and df from part a)
# Calculate p-value for two-tailed t-test
p_value = 2 * (1 - stats.t.cdf(np.abs(t_stat), df))  # 双侧检验需乘2

print(f'p-value: {p_value:.3f}')

p-value: 0.127


### Interpretation of p-value
The p-value is 0.127, which is greater than the significance level α = 0.05.

This indicates that if the null hypothesis (true mean = 10.0 cm) is true, there is a 12.7% probability of observing a test statistic as extreme as (or more extreme than) the calculated t-statistic (1.579).

Since the p-value > α, we fail to reject the null hypothesis, confirming our conclusion from part (a).

# Q2：

## (a) Compute the Risk Function $R(\theta, d)$
The risk function is the expectation of the loss function over the data: $R(\theta, d) = \mathbb{E}[L(\theta, d(x)) | \theta]$

Risk Function Calculation:
When θ=0:
R(0, d) = 2*(1-d(0))*(2/3) + 2*(1-d(1))*(1/3)
        = (4/3)*(1-d(0)) + (2/3)*(1-d(1))
        = 2 - (4/3)d(0) - (2/3)d(1)

When θ=1:
R(1, d) = 3d(0)*(3/4) + 3d(1)*(1/4)
        = (9/4)d(0) + (3/4)d(1)


## (b) Derive the Bayes Risk $r(\pi, d)$
The Bayes risk is the prior expectation of the risk function: $r(\pi, d) = \pi(0)R(0, d) + \pi(1)R(1, d)$

Bayes Risk Calculation:
r(π, d) = p*R(0,d) + (1-p)*R(1,d)
        = p*[2 - (4/3)d(0) - (2/3)d(1)] + (1-p)*[(9/4)d(0) + (3/4)d(1)]
        = 2p + [-4p/3 + 9(1-p)/4]d(0) + [-2p/3 + 3(1-p)/4]d(1)

Coefficient Simplification:
A = (-4p/3 + 9(1-p)/4) = (27 - 43p)/12
B = (-2p/3 + 3(1-p)/4) = (9 - 17p)/12
C = 2p

Thus: r(π, d) = (27-43p)/12 * d(0) + (9-17p)/12 * d(1) + 2p
